client/upe-onboarding-wizard/upe-preview-methods-selector/__tests__/setup-complete-task.test.js

```diff
@@ -1,52 +1,148 @@
 		);
-	};
 
-	it( 'Clicking "Add payment methods" should redirect back to settings page', async () => {
-		const setCompletedMock = jest.fn();
-		renderHelper( setCompletedMock );
-		expect( setCompletedMock ).not.toHaveBeenCalled();
+		expect( screen.getByText( /Setup complete/ ) ).toHaveTextContent(
+			'Setup complete!'
+		);
+	} );
+
+	it( 'renders setup complete messaging when context value says that one payment method has been removed', () => {
+		useEnabledPaymentMethodIds.mockReturnValue( [
+			[ 'card', 'sofort' ],
+			() => null,
+		] );
+		render(
+			<WizardContext.Provider
+				value={ {
+					completedTasks: {
+						'add-payment-methods': {
+							initialMethods: [ 'card', 'giropay', 'sofort' ],
+						},
+					},
+				} }
+			>
+				<WizardTaskContext.Provider value={ { isActive: true } }>
+					<SetupComplete />
+				</WizardTaskContext.Provider>
+			</WizardContext.Provider>
+		);
 
-		expect( screen.getByText( 'Go to Stripe settings' ).href ).toContain(
-			'admin.php?page=wc-settings&tab=checkout&section=stripe'
+		expect( screen.getByText( /Setup complete/ ) ).toHaveTextContent(
+			'Setup complete!'
 		);
 	} );
 
-	it( 'should show the 1 extra payment is selected compared to step 1', async () => {
-		renderHelper();
-		expect(
-			screen.getByText(
-				'Setup complete! 1 new payment method is now live on your store!'
-			)
-		).toBeInTheDocument();
+	it( 'renders setup complete messaging when context value says that one payment method has been added', () => {
+		useEnabledPaymentMethodIds.mockReturnValue( [
+			[ 'card', 'giropay' ],
+			() => null,
+		] );
+		render(
+			<WizardContext.Provider
+				value={ {
+					completedTasks: {
+						'add-payment-methods': {
+							initialMethods: [ 'card' ],
+						},
+					},
+				} }
+			>
+				<WizardTaskContext.Provider value={ { isActive: true } }>
+					<SetupComplete />
+				</WizardTaskContext.Provider>
+			</WizardContext.Provider>
+		);
+
+		expect( screen.getByText( /Setup complete/ ) ).toHaveTextContent(
+			'Setup complete! One new payment method is now live on your store!'
+		);
 	} );
 
-	it( 'should have 2 icons displaying after setup is completed', async () => {
-		renderHelper();
-		expect( screen.getAllByRole( 'img' ).length ).toBe( 2 );
+	it( 'renders setup complete messaging when context value says that more than one payment method has been added', () => {
+		useEnabledPaymentMethodIds.mockReturnValue( [
+			[ 'card', 'giropay', 'sofort', 'sepa_debit' ],
+			() => null,
+		] );
+		render(
+			<WizardContext.Provider
+				value={ {
+					completedTasks: {
+						'add-payment-methods': {
+							initialMethods: [ 'card' ],
+						},
+					},
+				} }
+			>
+				<WizardTaskContext.Provider value={ { isActive: true } }>
+					<SetupComplete />
+				</WizardTaskContext.Provider>
+			</WizardContext.Provider>
+		);
+
+		expect( screen.getByText( /Setup complete/ ) ).toHaveTextContent(
+			'Setup complete! 3 new payment methods are now live on your store!'
+		);
```

Comment:
Add a test case for [this scenario](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1887/files?file-filters%5B%5D=.js&hide-deleted-files=true#r711916678) may be.

Reply:
There are the "should not render setup complete messaging when context value is undefined" and "renders setup complete messaging when context value is `true`" cases for those scenarios - but since now we're using optional chaining, I'll remove them 👍 (since you can't use optional chaining on `true`)